src/v/kafka/client/client.cc

```diff
@@ -289,10 +289,13 @@ ss::future<produce_response> client::produce_records(
             std::move(*p.records->adapter.batch));
       });
 
+    chunked_vector<topic_produce_response> responses_cv;
```

Comment:
nit: Is there something preventing us from using list-initialization for the chunked_vector like we did previously with the vector?

Reply:
Right you are `std::initializer_list` only returns constant iterators. Dunno how I missed it before. Definitely something we should avoid using if possible.